source/common/protobuf/utility.cc

```diff
@@ -197,12 +198,7 @@ void deprecatedFieldHelper(Runtime::Loader* runtime, bool proto_annotated_as_dep
   if (warn_only) {
     ENVOY_LOG_MISC(warn, "{}", with_overridden);
   } else {
-    const char fatal_error[] =
-        " If continued use of this field is absolutely necessary, see "
-        "https://www.envoyproxy.io/docs/envoy/latest/configuration/operations/runtime"
-        "#using-runtime-overrides-for-deprecated-features for how to apply a temporary and "
-        "highly discouraged override.";
-    throw ProtoValidationException(with_overridden + fatal_error, message);
+    validation_visitor.onDeprecatedField("type " + message.GetTypeName() + " " + with_overridden);
```

Comment:
I think this should always be called, even if warning, but we can add a parameter to control whether it's expected to log or throw an exception (`warn_only`), maybe call it `soft_deprecation`?